thunder/core/transform_common.py

```diff
@@ -331,20 +331,16 @@ def thunder_140410131706304(x, y):
     return cse_trace
 
 
-# Base class for all types of Transform.
-class Transform:
-    pass
-
-
-# Below are the types of Transform that user can create and apply to the `jitted` function.
-class EarlyTransform(Transform, ABC):
-    """
-    EarlyTransform enables transforming prologue, computation and epilogue trace.
-    Note that the computation trace here is before the autograd transform, so any update to
-    the computation trace will also update backward trace.
-    """
-
-    def transform_traces(self, prologue_trace: Trace, computation_trace: Trace, epilogue_trace: Trace | None, **kwargs):
+# The transform class, at different points in the processing, the methods are executed. Typically, a subset of them is implemented.
+class Transform(ABC):
+    def transform_traces_early(
```

Comment:
We could adopt a name that is more explicit. Like `pre-<something>` or `post-<something>`.
Like `post-optimization` for the other method is a good one IMO. It will probably make names more future-proof.

Lightning uses `on_before_<event>` and `on_after_<event>` like `on_before_zero_grad` or `on_after_backward`, and it served us well.

Reply:
Fair point for early: pre_prologue ?

for additionally I would not sweat it because we want to get rid of it.